packages/broker/src/helpers/parser.ts

```diff
@@ -0,0 +1,35 @@
+import { Request } from 'express'
+
+export const parsePositiveInteger = (n: string) => {
+    const parsed = parseInt(n)
+    if (!Number.isInteger(parsed) || parsed < 0) {
+        throw new Error(`${n} is not a valid positive integer`)
+    }
+    return parsed
+}
+
+export const parseTimestamp = (millisOrString: number|string) => {
+    if (typeof millisOrString === 'number') {
+        return millisOrString
+    }
+    if (typeof millisOrString === 'string') {
+        // Try if this string represents a number
+        const timestamp = Number(millisOrString) || Date.parse(millisOrString)
+        if (Number.isNaN(timestamp)) {
+            throw new Error(`Invalid timestamp: ${millisOrString}`)
+        } else {
+            return timestamp
+        }
+    } else {
+        throw new Error(`Invalid timestamp: ${millisOrString}`)
+    }
+}
+
+export const getQueryParameter = (name: string, req: Request, parser?: (input: string) => any) => {
```

Comment:
(This change might break compilation)
```suggestion
export const getQueryParameter = <T>(name: string, req: Request, parser?: (input: string) => T): T | string | undefined => {
```

Reply:
Fixed, removed use case where parser is not defined